### ISSUE TITLE
Privacy - iOS: JitsiMeet.framework utilizes react-native-netinfo which sends HTTP requests to Google

### ISSUE THREAD

**Issue by Letterus** [jitsi/jitsi-meet#6474]
## Description

JitsiMeet.framework is used by the [iOS Riot app](https://github.com/vector-im/riot-ios). Using Riot it was discovered that the app sends HTTP requests to Google. This is unwanted due to privacy concerns. It seems the requests emerge from react-native-netinfo which is utilized by JitsiMeet.framework.

## Current behavior

JitsiMeet.framework sends (multiple) request to ```clients3.google.com/generate_204``` during the lifetime of the app.

## Expected Behavior

JitsiMeet.framework should not loose data to Google. If react-native-netinfo provides desireable features this behaviour should be optional/configurable.

## Possible Solution

Update to a version of react-native-netinfo where this behaviour is configurable.

## Steps to reproduce

Install the iOS Riot app and configure it. Install the Charles proxy app. Start the Riot app. You'll see the request to clients3.google.com in the log of Charles.

# Environment details

iOS 13.4, iOS Riot app

**Comment by saghul** [comment]
Thanks for the report @Letterus ! I think we could just use meet.jit.si and since it uses a HEAD request just check for 200. Does that sound ok?

**Comment by saghul** [comment]
Yeah, the plan would be to use the updated version to set meet.jit.si as the target.

> From a legal point of view (regarding European GDPR) this does not really help, because every 3rd party host different to your service host is one you have to control (from a legal perspective). And you will have to declare its usage.

Not sure I understand, you mean this in the context of people using their own deployment? I think we can try to make it work with the configured server instead. As long as a HEAD request returns 200 we are good. It's not part of any critical functionality, at any rate, so if a server is misconfigured and fails, it's ok.

Does that sound better?

**Comment by Letterus** [comment]
Yeah, sounds nice. And yes: If you set the configured Jitsi server as default it could be used nicely in the context of other deployments (like Riot). Thank you!

**Comment by saghul** [comment]
This should be gone now (in master), since netinfo uses the SCNetworkReachability API now.

**Comment by saghul** [comment]
No configuration will be necessary.

The update is not part of SDK 4.1.0, it la fed afterwards and will be part of SDK 5.

You could try building master to help us double check, os test the beta releases through TestFlight.

**Comment by csett86** [comment]
I retested this with current beta from TestFlight (22.0.0(157)) and unfortunately the requests to clients3.google.com are still there. See trace from Charles below.

![IMG_3377](https://user-images.githubusercontent.com/1392875/152852124-5b255833-fedc-4deb-b04b-2472679e0520.jpeg)
![IMG_3378](https://user-images.githubusercontent.com/1392875/152852139-0609aa9e-a67f-406b-a622-6858aac345d3.jpeg)

I also checked that the requests were really coming from Jitsi Meet by checking the User-Agent in the request headers:

![IMG_3379](https://user-images.githubusercontent.com/1392875/152852143-a6e56df5-bf80-4374-8230-d704c42c8a72.jpeg)

**Comment by csett86** [comment]
The code in the react native netinfo module also does not refer to any native code, its pure typescript:

https://github.com/react-native-netinfo/react-native-netinfo/blob/master/src/internal/internetReachability.ts

**Comment by saghul** [comment]
Thanks for taking a look! Gotta dive deeper I guess.

The iOS part also has this: https://github.com/react-native-netinfo/react-native-netinfo/blob/master/ios/RNCConnectionState.m which uses the reahability API. Maybe there is a way to enable that.

**Comment by saghul** [comment]
At a gland looks like setting `reachabilityShouldRun` should so, this way only updates from the native side will be considered. We should also set the URL to something invalid, for safety.

**Comment by csett86** [comment]
From my understanding the native code covers `type`, `isConnected` and `details` from the `NetInfoState`, while `isInternetReachable` is only provided via the `fetch()` to clients3.google.com.

Setting `reachabilityShouldRun` to a function returning false (`() => false`) disables the internet connectivity check, but leaves everything else like the type and details in place. If this would be ok, I would be happy to provide a PR for this.

**Comment by saghul** [comment]
> From my understanding the native code covers `type`, `isConnected` and `details` from the `NetInfoState`, while `isInternetReachable` is only provided via the `fetch()` to clients3.google.com.

Ah crap, you're mostly right. Looks like on Android the native side an also provide that, but not iOS.

> Setting `reachabilityShouldRun` to a function returning false (`() => false`) disables the internet connectivity check, but leaves everything else like the type and details in place. If this would be ok, I would be happy to provide a PR for this.

No, that kinda botches the current functionality :-/ I think the change that we ought to do here is to direct the HEAD request at the config.js file of the configured server. That should be a good enough indicator I think.

**Comment by pixlwave** [comment]
Thought it was worth mentioning in this thread that upgrading Jitsi to 5.0.2 in Element iOS has stopped the app from pinging Google so long as you don't join a Jitsi call. Once that happens then we see the checks every minute even once you have left the call.t

> I think the change that we ought to do here is to direct the HEAD request at the config.js file of the configured server.

Checking meet.element.io/config.js, the file is 60KB which could be considered a bit heavy for a mobile device when there isn't much signal. Would it be feasible to expose a property so that SDK users could configure the URL themselves?

**Comment by saghul** [comment]
> Checking meet.element.io/config.js, the file is 60KB which could be considered a bit heavy

A HEAD request wouldn't download the content.

**Comment by pixlwave** [comment]
Ohhh TIL, sounds perfect 👍
